extensions/kafka-streams/runtime/src/main/java/io/quarkus/kafka/streams/runtime/KafkaStreamsProducer.java

```diff
@@ -54,6 +54,8 @@
     private static final Logger LOGGER = Logger.getLogger(KafkaStreamsProducer.class.getName());
     private static volatile boolean shutdown = false;
 
+    public static volatile boolean ignoreShutdownFlag = false;
```

Comment:
Instead of having to reason about the two booleans, how about establishing an enum which exposes all the potential states the producer can be in, e.g. `enum State { RUNNING, RUNNING_NON_STOPPABLE, STOPPED}`?

Reply:
I'll drop the 2nd boolean